detections/ec2_excessive_runinstances_mltk.yml

```diff
@@ -0,0 +1,97 @@
+asset_type: AWS Instance
+confidence: medium
+creation_date: '2019-11-14'
+data_metadata:
+  data_source:
+    - AWS CloudTrail logs
+  data_sourcetypes:
+    - aws:cloudtrail
+  providing_technologies:
+    - AWS
+description: This search looks for CloudTrail events where a user successfully launches
+  an abnormally high number of instances.
+detect:
+  splunk:
+    correlation_rule:
+      macros:
+        - ec2_excessive_runinstances_mltk_input_filter
+      notable:
+        nes_fields: userName
+        rule_description: An abnormally high number of instances were launched by
+          a user within in a 10-minute window
+        rule_title: High Number of instances launched by $userName$
+      risk:
+        risk_object: userName
+        risk_object_type:
+          - user
+        risk_score: 30
+      schedule:
+        cron_schedule: '*/10 * * * *'
+        earliest_time: -70m@m
+        latest_time: -10m@m
+      search: sourcetype=aws:cloudtrail eventName=RunInstances errorCode=success | bucket span=10m _time | stats sum(instances_launched) as instances_launched by _time userName | apply ec2_excessive_runinstances_v1 | rename "IsOutlier(instances_launched)" as isOutlier | where isOutlier=1
```

Comment:
use datamodels and embed macros

Reply:
I've added the input filter macro in my last commit. Dorsey mentioned that we could do a separate AWS and generic cloud analytics stories for this.